docs/designers-developers/readme.md

```diff
@@ -1,11 +0,0 @@
-# Designer & Developer Handbook
-
-The Gutenberg project is transforming the way content is created on WordPress. A block editor was the first product launched creating a new methodology for working with content. This handbook provides documentation for how designers and developers can extend the editor.
```

Comment:
Note that the current handbook sync [has been disabled](https://meta.trac.wordpress.org/ticket/4244#comment:7) so this removal won't trigger any update (nor the other minor content updates).